src/server/builder.ts

```diff
@@ -223,7 +223,7 @@ namespace ts.server {
             for (const reference of this.references) {
                 reference.removeReferencedBy(this);
             }
-            this.references = createSortedArray<ModuleBuilderFileInfo>();
+            clear(this.references);

```

Comment:
Looks like that's safe -- `this.references` is owned by the `ModuleBuilderFileInfo` and is only temporarily borrowed elsewhere, but a reference to it isn't stored.